code/modules/mob/living/carbon/human/species/slime.dm

```diff
@@ -82,9 +86,17 @@
 	if(recolor)
 		recolor.Remove(H)
 	REMOVE_TRAIT(H, TRAIT_WATERBREATH, "species")
+	UnregisterSignal(H, COMSIG_HUMAN_UPDATE_DNA)
+
+/datum/species/slime/proc/blend(mob/living/carbon/human/H)
+	if(H.blood_color == null || H.blood_color != BlendRGB(H.skin_colour, "#acacac", 0.5)) // Put here, so if it's a roundstart, dyed, or CMA'd slime, their blood changes to match skin
+		if(H.blood_color == null)
+			H.blood_color = H.skin_colour //Just to get it not null so the math works
+		H.blood_color = BlendRGB(H.skin_colour, "#acacac", 0.5) // Blends this to make it work better
```

Comment:
```suggestion
	var/new_color = BlendRGB(H.skin_colour, "#acacac", 0.5) // Blends this to make it work better
	if(H.blood_color != new_color) // Put here, so if it's a roundstart, dyed, or CMA'd slime, their blood changes to match skin
		H.blood_color = new_color
```
`H.blood_color` is never the same as the new color if it's `null`. Caching the `new_color` will save on a call. It's the small things

Setting `H.blood_color` on line 94 doesn't do much since you instantly change it on line 95. The color you use in there is also the `H.skin_colour`.